packages/core/src/modules/credentials/formats/jsonld/JsonLdCredentialFormatService.ts

```diff
@@ -317,20 +300,17 @@ export class JsonLdCredentialFormatService extends CredentialFormatService<JsonL
   ): Promise<void> {
     const credentialAsJson = attachment.getDataAsJson<W3cVerifiableCredential>()
     const credential = JsonTransformer.fromJSON(credentialAsJson, W3cVerifiableCredential)
-    MessageValidator.validateSync(credential)
 
     // compare stuff in the proof object of the credential and request...based on aca-py
 
-    const requestAsJson = requestAttachment.getDataAsJson<SignCredentialOptionsRFC0593>()
-    const request = JsonTransformer.fromJSON(requestAsJson, JsonLdCredentialDetail)
+    const requestAsJson = requestAttachment.getDataAsJson<JsonLdSignCredentialFormat>()
 
     if (Array.isArray(credential.proof)) {
-      // question: what do we compare here, each element of the proof array with the request???
       throw new AriesFrameworkError('Credential arrays are not supported')
     } else {
       // do checks here
-      this.compareCredentialSubject(credential, request.credential)
-      this.compareProofs(credential.proof, request.options)
+      this.compareCredentialSubject(credential, requestAsJson)
+      this.compareProofs(credential.proof, requestAsJson)
     }
 
     const verifiableCredential = await this.w3cCredentialService.storeCredential(agentContext, {
```

Comment:
This doesn't yet verify the credential as described in #1132 (the PR body mentions it fixes this issue).

Reply:
I assume this should be a call to the w3cCredentialService `verifiyCredential()` method.